dace/transformation/dataflow/map_fusion.py

```diff
@@ -480,6 +480,10 @@ def fuse_nodes(self, sdfg, graph, edge, new_dst, new_dst_conn, other_edges=None)
             else:
                 local_node = edge.src
                 src_connector = edge.src_conn
+                if isinstance(edge.src, nodes.AccessNode):
```

Comment:
From the PR description, I understand that this fixes a case where the memlet is left carrying an "obsolete" transient scalar from a (previous?) fusion. The `__s0_...` name is characteristic of that. This is indeed the correct way to fix the issue. However, I would place a few more safeguards (or, if they are already satisfied, put a note to explain how and why they are satisfied for future reference). Namely:
- Is this change really needed, i.e., does the memlet indeed match neither the source nor the destination target?
- Is the priority order programmed (src -> dst) always correct if a change is needed? Does the (other) subset need to be checked? What about `Memlet._is_data_src`? The change above is definitely correct if both the src and dst are scalars. However, they could be arrays (for example, think of the "strange" subgraphs created by SubgraphFusion; you may have already had issues with them before).

If this is blocking and urgent, my suggestion would be to narrow the application of the above fix as follows:
```python
test_data = []
test_data.append(edge.src.data if isinstance(edge.src, nodes.AccessNode) else None)
test_data.append(edge.dst.data if isinstance(edge.dst, nodes.AccessNode) else None)
current_data = edge.data.data
if current_data is not None and current_data not in test_data and list(edge.data.subset) == [0]:
    for new_data in test_data:
        if new_data is not None and isinstance(sdfg.arrays[new_data], dace.data.Scalar):
            edge.data.data = new_data
````
The above code is indicative and not tested (you don't need the None values in test data unless you want to know if src or dst was selected), but I think it resolves your issue while avoiding triggering in cases that have not been studied yet.

Reply:
Thank you for a quick and informative reply. I will look in detail and try your suggestion. I just wanted to quickly answer to the question mark about the origin of the transient scalar:
> from a (previous?) fusion

No, the transient scalar originates from the same fusion step, just a few lines above, at line 463:
```
            edge.data.data = local_name
            edge.data.subset = "0"
```